src/main/java/ui/components/pickers/PickerMilestone.java

```diff
@@ -0,0 +1,179 @@
+        Label milestone = new Label(getTitle());
+        adjustWidthToFont(milestone);
+        return milestone;
+    }
+
+    private void adjustWidthToFont(Label milestone) {
+        FontLoader fontLoader = Toolkit.getToolkit().getFontLoader();
+        double width = fontLoader.computeStringWidth(milestone.getText(), milestone.getFont());
+        milestone.setPrefWidth(width + 30);
+        milestone.getStyleClass().add("labels");
+    }
+
+    private Label createCustomLabel(int fontSize) {
+        Label milestone = new Label(getTitle());
+        milestone.setFont(new Font(fontSize));
+        adjustWidthToFont(milestone);
+        return milestone;
+    }
+
+    private void setOpenStatusColour(Label milestone) {
+        milestone.setStyle("-fx-background-color: " + (isOpen() ? OPEN_COLOUR : CLOSE_COLOUR) + ";");
+    }
+
+    private void setFadedInUI(Label milestone) {
+        milestone.setStyle(milestone.getStyle() + "-fx-opacity: 40%;");
+    }
+
+    private void setHighlightedInUI(Label milestone) {
+        milestone.setStyle(milestone.getStyle() + "-fx-border-color: black;");
+    }
+
+    private void setSelectedInUI(Label milestone) {
+        milestone.setText(milestone.getText() + " ✓");
+    }
+
+    private void setRemovedInUI(Label milestone) {
+        milestone.getStyleClass().add("labels-removed"); // add strikethrough
+    }
+
+    public Node getNewlyAssignedMilestoneNode(boolean hasSuggestion) {
+        Label milestone = createCustomLabel(BIG_LABEL_FONT);
+        setOpenStatusColour(milestone);
+
+        if (hasSuggestion) {
+            setFadedInUI(milestone);
+            if (isSelected) setRemovedInUI(milestone);
+        }
+
+        if (isSelected) {
+            setHighlightedInUI(milestone);
+        }
+
+        return milestone;
+    }
+
+    public void setSelected(boolean isSelected) {
+        this.isSelected = isSelected;
+    }
+
+    public boolean isSelected() {
+        return this.isSelected;
+    }
+
+    public void setHighlighted(boolean isHighlighted) {
+        this.isHighlighted = isHighlighted;
+    }
+
+    public boolean isHighlighted() {
+        return this.isHighlighted;
+    }
+
+    public void setFaded(boolean isFaded) {
+        this.isFaded = isFaded;
+    }
+
+    public boolean isFaded() {
+        return this.isFaded;
+    }
+    public void setExisting(boolean isExisting) {
+        this.isExisting = isExisting;
+    }
+
+    public boolean isExisting() {
+        return this.isExisting;
+    }
+
+    /**
+     * Milestones which are open are deemed to be smaller than milestones which are closed
+     * Milestones with earlier due dates are deemed to be "larger".
+     *
+     * In addition, this treats null milestones with no set due date to be "smaller than"
+     * those which have due dates
+     *
+     * @param milestone
+     * @return
+     */
+    @Override
+    public int compareTo(PickerMilestone milestone) {
+        if (isHighlighted) return -1;
```

Comment:
You're right that here it will probably never matter, but I'd lean towards a proper compareTo. Mostly so there are no surprises, and so we don't have to go to the trouble of maintaining that invariant (which is outside this class). Seems you've changed this also. :stuck_out_tongue: